dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESMappingUtilHelper.java

```diff
@@ -298,6 +301,61 @@ private void addMappingForFieldIfNeeded(final String indexName,
         }
     }
 
+    /**
+     * Given a {@link Field}, obtains the ES mapping according to the {@link Field} type
+     * @param field
+     * @param fieldVariableName
+     * @return
+     */
+    private String getMappingForField(final Field field, final String fieldVariableName) {
+        String mappingForField = null;
+        if (field instanceof DateField || field instanceof DateTimeField
+                || field instanceof TimeField) {
+            mappingForField = "{\n\"type\":\"date\",\n";
+            mappingForField += "\"format\": \"yyyy-MM-dd't'HH:mm:ss||MMM d, yyyy h:mm:ss a||yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis\"\n}";
+        } else if (field instanceof TextField || field instanceof TextAreaField
+                || field instanceof WysiwygField || field instanceof RadioField
+                || field instanceof SelectField) {
+            if (field.dataType() == DataTypes.BOOL) {
+                mappingForField = "{\n\"type\":\"boolean\"\n}";
+            } else if (field.dataType() == DataTypes.FLOAT) {
+                mappingForField = "{\n\"type\":\"float\"\n}";
+            } else if (field.dataType() == DataTypes.INTEGER) {
+                mappingForField = "{\n\"type\":\"integer\"\n}";
+            } else if (!matchesExclusions(fieldVariableName)){
+                mappingForField = "{\n\"type\":\"text\"\n}";
+            }
+        }
+        return mappingForField;
+    }
+
```

Comment:
codacy already recommended using StringBuilder here and I like that but I think is more maintainable doing something like :

mappingForField =  String.format("{\n\"type\":\"%s\"\n}",fieldType);    

and pass fieldType. That way we can have that string format declared just once.

and the fieldType can be returned out of function so the function I imagine could use a static map and look like this:

```
ImmutableMap.of(
DataTypes.BOOL,DataTypes.BOOL.toString().toLowercase(),   //Could do something like this 
DataTypes.FLOAT, "float",   //or simply
DataTypes.INTEGER, "integer"
) ;
```
I've seen that Using maps like this to eliminate nested ifs. Is called the anti-if pattern.

Reply:
Good suggestion. Done!